src/mui/input/PhoneInput.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Field } from 'redux-form';
+import { asYouType } from 'libphonenumber-js';
+
+import TextInput from './TextInput';
+
+export const parseInput = number => {
+    if (!number) {
+        return '';
+    }
+
+    if (number.charAt(0) === '0') {
+        const rest = number.slice(1);
+        number = `+33${rest}`;
+    }
```

Comment:
```js
 var a = require('libphonenumber-js');
 var p = new a.asYouType('FR');
 p('0623576067');
```

Reply:
this is working

Reply:
Damn it the lib doesn't handle intl output

Reply:
Ok, first there is a bug in libphonenumber:
* in 'as you type.js' line 403 the country_code is set to undefined because metadata are array an not object.
Once the fix will be done, the nice way to handle what we need would be to do either:
* Define a custom class and overwrite full_phone_number behavior
* Do a PR to handle an output format in asYouType constructor and thus adapt full_phone_number

Reply:
I just push the PR.

Reply:
https://github.com/catamphetamine/libphonenumber-js/pull/149

Reply:
Already merged and published by the maintener